File: TeamCubing.Domain/Models/Solve.cs
using Newtonsoft.Json;
using TeamCubing.Domain.DTO;

namespace TeamCubing.Domain.Models;

public class Solve
{
    [JsonProperty("solveNumber")]
    public int SolveNumber { get; set; }

    [JsonProperty("scramble")]
    public string Scramble { get; set; }

    [JsonProperty("scrambledPuzzleImage")]
    public PuzzleImage ScrambledPuzzleImage { get; set; }

    [JsonProperty("startTime")]
    public DateTime StartTime { get; set; }

    [JsonProperty("results")]
    public List<SolveResult> Results { get; set; } = new();
}

File: src/TeamCubing.API/ClientApp/src/app/models/solve.ts
import { PuzzleImage } from './puzzles/puzzleImage';

export interface Solve {
  solveNumber: number;
  results: SolveResult[];
  startTime: Date;
  scramble: string;
  scrambledPuzzleImage: PuzzleImage;
}

export interface SolveResult extends BaseSolveResult {
  userName: string;
}

export interface BaseSolveResult {
  time: number;
  penalty: Penalty;
}

export enum Penalty {
  NoPenalty,
  PlusTwo,
  DNF,
}
